File: lib/app/data/models/dto/create_event_dto.dart
class CreateEventDto {
  final String title;
  final String description;
  final DateTime startDate;
  final DateTime endDate;
  final String? picturePath;

  CreateEventDto(
      {required this.title,
      required this.description,
      required this.startDate,
      required this.endDate,
      this.picturePath});

  Map<String, dynamic> toJson() => {
        "title": title,
        "description": description,
        "startDate": startDate.toIso8601String(),
        "endDate": endDate.toIso8601String(),
        if (picturePath != null) "picturePath": picturePath
      };
}

File: src/modules/events/dto/create-event.dto.ts
import { ApiProperty } from "@nestjs/swagger";
import { IsNotEmpty } from "class-validator";
import { CreationEntityDto } from "src/common/dto/creation.dto";
import { Company } from "src/modules/company/entities/company.entity";

export class CreateEventDto extends CreationEntityDto {
    @ApiProperty()
    @IsNotEmpty()
    title:string
    @ApiProperty()
    @IsNotEmpty()
    description:string
    @ApiProperty()
    @IsNotEmpty()
    startDate:Date
    @ApiProperty()
    endDate:Date
    @ApiProperty()
    pathPicture
    @ApiProperty()
    location :string
    company :Company
    


}
